### ISSUE TITLE
Allow to publish shared presets with a renovate.json file instead of in package.json

### ISSUE THREAD

**Issue by felixfbecker** [renovatebot/renovate#2312]
**What would you like Renovate to be able to do?**
I would like to use the JSON schema validation and autocompletion when writing my shared preset (https://github.com/renovatebot/renovate/issues/545)
But if the config is in package.json, that doesn't work, because it's not part of the package.json schema.

**Describe the solution you'd like**

Instead, I would like to add a `main` field in package.json that points to the renovate.json file that represents the preset. This is how TSLint resolves `extends` fields, it simply resolves them like a node module would be.

**Comment by rarkins** [comment]
I will implement this using GitHub-hosted presets only at first, hence this is waiting on #2102. Otherwise we need to be pulling down tarballs from npmjs and extracting the `renovate.json`.

**Comment by felixfbecker** [comment]
If you don't want to pull tarballs you could also use unpkg.com - like https://unpkg.com/@sourcegraph/tslint-config/tslint.json

**Comment by rarkins** [comment]
I’m a but hesitant to rely on them because if they have a failure we need to bail on the whole run to make sure we don’t do wacky stuff (eg close or create PRs that shouldn’t be there). Npmjs is already challenging enough but I might reconsider due to changelog files too (very hard to locate in monorepos but easy to pluck from unpkg).
